### PR TITLE
refactor(client): test grouping

### PR DESCRIPTION
**Pull request by danstarns**
The functional test directory was getting large and with us adding more and more tests to it… It will only get larger. This refactor adds grouping to the functional tests so that related tests are within the same directory. For example: 

Before

1. `queryRaw-send-type-hits/`
2. `queryRaw-typed-results/`
3. `queryRaw-types-results-advanced-and-native`/

Now: 

1. `queryRaw/`
    - `send-type-hints/`
    - `typed-results/`
    - `typed-results-advanced-and-native/`

I also added an issues directory where I have moved some existing tests that were replicating and adding a regression for a particular Github issue. I have renamed those tests after the issue number, as I found when adding a regression test for a particular issue it was very hard to name them and so if they are named after the issue at least we can use that as the context.

This PR diff is rather large and so it's best to checkout the branch and manually sees the changes. I have grouped the following: 

1. issues
2. methods
3. metrics
4. queryRaw
5. composites
6. binary-engine

### CODE REVIEW

**Comment by danstarns** [comment]
This was originally discussed here: https://github.com/prisma/prisma/pull/14831#discussion_r946798091

**Review by SevInf — APPROVED** [review]
Not sure why MySQL test is failing (did docker image got updated?), but code-wise LGTM.
Great improvement, it will help in field references PR as well.

File: TESTING.md [hunk 1]
```diff
@@ -193,6 +193,24 @@ Test consists of the 3 files:
...
+
+    - test.ts
+    - \_matrix.ts
+    - `prisma/_schema.ts`
+
+2.  `named/`
+
+    - test.ts
+    - \_matrix.ts
+    - `prisma/_schema.ts`
```
Comment (aqrln) [inline TESTING.md:212]: ditto

File: packages/client/tests/functional/issues/11233/tests.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import testMatrix from './_matrix'
 declare let prisma: PrismaClient
 declare let Prisma: typeof PrismaNamespace
 
+// https://github.com/prisma/prisma/issues/11233
```
Comment (aqrln) [inline packages/client/tests/functional/issues/11233/tests.ts:9]: Look like an accidental change, the issue behind the link is irrelevant.

Comment (aqrln) [reply packages/client/tests/functional/issues/11233/tests.ts:9]:
```suggestion
```

**Review by aqrln — APPROVED** [review]
This is a very nice improvement!

File: packages/client/tests/functional/issues/11233/tests.ts (hunk 2 above)
Comment (danstarns) [reply packages/client/tests/functional/issues/11233/tests.ts:9]: Are you sure, the test is "should not throw when using Prisma.empty inside $executeRaw" and the issue "Cannot pass Prisma.empty to $executeRaw function"

Comment (aqrln) [reply packages/client/tests/functional/issues/11233/tests.ts:9]: Sorry, my bad. The context got mixed up in my head while I was scrolling, I was looking at the wrong piece of code. Sorry again for the confusion!

Comment (aqrln) [reply packages/client/tests/functional/issues/11233/tests.ts:9]: (this file literally has the issue number in its name 😅)

File: TESTING.md [hunk 3]
```diff
@@ -193,6 +193,24 @@ Test consists of the 3 files:
...
+
+If you have related tests but different schemas, you can nest directories. For example, composite indexes have the following file structure:
+
+`composite-index/`
+
+1.  `list/`
+
+    - test.ts
+    - \_matrix.ts
+    - `prisma/_schema.ts`
```
[the remark this answers is not in the document]
Comment (danstarns) [reply TESTING.md]: https://github.com/prisma/prisma/pull/14988/commits/a4156a59cfc576e317a0310410bdf7ea777240b3

**Pull request merged**